smack-experimental/src/main/java/org/jivesoftware/smackx/muclight/MultiUserChatLight.java

```diff
@@ -0,0 +1,555 @@
+     *         message becoming available.
+     * @throws InterruptedException
+     */
+    public Message nextMessage(long timeout) throws InterruptedException {
+        return messageCollector.nextResult(timeout);
+    }
+
+    /**
+     * Adds a stanza(/packet) listener that will be notified of any new messages
+     * in the group chat. Only "group chat" messages addressed to this group
+     * chat will be delivered to the listener.
+     *
+     * @param listener
+     *            a stanza(/packet) listener.
+     * @return true if the listener was not already added.
+     */
+    public boolean addMessageListener(MessageListener listener) {
+        return messageListeners.add(listener);
+    }
+
+    /**
+     * Removes a stanza(/packet) listener that was being notified of any new
+     * messages in the MUCLight. Only "group chat" messages addressed to this
+     * MUCLight were being delivered to the listener.
+     *
+     * @param listener
+     *            a stanza(/packet) listener.
+     * @return true if the listener was removed, otherwise the listener was not
+     *         added previously.
+     */
+    public boolean removeMessageListener(MessageListener listener) {
+        return messageListeners.remove(listener);
+    }
+
+    /**
+     * Remove the connection callbacks used by this MUC Light from the
+     * connection.
+     */
+    private void removeConnectionCallbacks() {
+        connection.removeSyncStanzaListener(messageListener);
+        if (messageCollector != null) {
+            messageCollector.cancel();
+            messageCollector = null;
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "MUC Light: " + room + "(" + connection.getUser() + ")";
+    }
+
+    /**
+     * Create new MUCLight.
+     * 
+     * @param roomName
+     * @param subject
+     * @param customConfigs
+     * @param occupants
+     * @return true if it was successfully created
+     * @throws NotConnectedException
+     * @throws XmppStringprepException
+     * @throws InterruptedException
+     */
+    public boolean create(String roomName, String subject, HashMap<String, String> customConfigs, List<Jid> occupants)
+            throws NotConnectedException, XmppStringprepException, InterruptedException {
+        MUCLightCreateIQ createMUCLightIQ = new MUCLightCreateIQ(room, roomName, occupants);
+
+        connection.addSyncStanzaListener(messageListener, fromRoomGroupchatFilter);
+        messageCollector = connection.createPacketCollector(fromRoomGroupchatFilter);
+
+        StanzaFilter responseFilter = new AndFilter(FromMatchesFilter.createBare(room.asDomainBareJid()),
+                new StanzaTypeFilter(IQ.class));
+
+        try {
+            IQ responseIq = connection.createPacketCollectorAndSend(responseFilter, createMUCLightIQ)
+                    .nextResultOrThrow();
+            return responseIq.getType().equals(IQ.Type.result);
```

Comment:
@Flowdalic Just to be sure I would keep `return responseIq.getType().equals(IQ.Type.result);` instead of `return true;`